File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} list1
     * @param {ListNode} list2
     * @return {ListNode}
     */
    mergeTwoLists(list1, list2) {
        if (!list1) {
            return list2
        }
        if (!list2) {
            return list1
        }
        let newHead = list1.val <= list2.val ? list1 : list2;
        if(list1.val <= list2.val){
           newHead =  list1
           list1 = list1.next
        }else{
            newHead = list2
            list2 = list2.next
        }
        let current = newHead;
        while(list1 && list2){
            if(list1.val <= list2.val){
                current.next =  list1
                current = list1
                list1 = list1.next
            }else{
                current.next =  list2
                current = list2
                list2 = list2.next
            }
        }
        current.next = list1 ? list1: list2

        
        return newHead;
    }
}

File: Data Structures & Algorithms/reverse-linked-list-ii/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        prev = None
        curr = head
        currIdx = 1
        while  currIdx < left:
            prev = curr
            curr = curr.next
            currIdx += 1

        startHead = curr
        startHeadPrev = prev
        # handle case of 1
        # print("hell", startHead.val, startHeadPrev)
        while currIdx <= right:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp
            currIdx += 1
        # curr
        # temp = curr.next
        # print("hell", prev.val)
        startHead.next = curr
        # head = prev 
        if startHeadPrev:
            startHeadPrev.next = prev
        else:
            head = prev
        # curr.next = startHeadPrev

        return head

        
        